atlp-rwanda/atlp-pulse-fn
# #575 Move client-side attendance logic to server

## PR Description

This PR moves the attendance tracking logic from the client-side to the server-side to improve performance, scalability, and data security. 

## Description of tasks that were expected to be completed
- Migrated client-side attendance tracking logic to the server.
- Created new API endpoints to handle attendance-related requests.
- A TTL is now able to record an attendance
- Updated front-end code to communicate with the new server endpoints.
- Ensured that the changes do not negatively impact any other functionality.

## How has this been tested?

- Clone this repository and navigate to the project directory
- Ensure You have backend service running and set your .env file using .env.example.
```bash
  git clone https://github.com/atlp-rwanda/atlp-pulse-fn.git
  cd atlp-pulse-fn
  git checkout fix-attendance-logic
  npm I
  npm run test
  or
  npm run dev
  ```
- Login in with valid TTL or coordinator credentials
- Navigate to attendance page

## Number of Commits

- 1 Commit

## Please check this Checklist before you submit your PR:
